onedal/neighbors/neighbors.py

```diff
@@ -131,20 +131,20 @@ def _get_onedal_params(self, X, y=None):
             "method": self._fit_method,
             "radius": self.radius,
             "class_count": class_count,
-            "neighbor_count": self.n_neighbors,
+            "neighbor_count": self.n_neighbors if n_neighbors is None else n_neighbors,
             "metric": self.effective_metric_,
             "p": self.p,
             "metric_params": self.effective_metric_params_,
             "result_option": "indices|distances" if y is None else "responses",
         }
 
-    def _get_daal_params(self, data):
+    def _get_daal_params(self, data, n_neighbors=None):
```

Comment:
use private attribute self._n_neighbors instead of extending this funcs signature

Reply:
This is necessary to prevent wrong values being stored in self.n_neighbors when using the fitted X values.

Reply:
@samir-nasibli I guess I don't understand. It would just match the value in self.n_neighbors in most cases, and then we would still have to if statement between self.n_neighbors and self._n_neighbors.  Could you explain a little more?